core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -122,7 +126,7 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
         } else {
           logInfo("Registered executor: " + executorRef + " with ID " + executorId)
           context.reply(RegisteredExecutor)
-
+          latestTokens.foreach(x => context.reply(NewTokens(x)))
```

Comment:
So this is weird. You just "replied" in the line above, so "replying" here doesn't make sense. Perhaps you mean `send`?

Also, why not stash the tokens in the `RegisteredExecutor` message? One less message to send!